lib/data_source/client/github.rb

```diff
@@ -7,40 +7,105 @@
         params = search_params(params[:query], modifiers)
         response = request('/search/repositories', params)
         handle_response(response)[:items]
       end
 
       def user_repos
         params = {
-          sort: 'pushed',
-          direction: 'desc',
-          per_page: 100
+            sort: 'pushed',
+            direction: 'desc',
+            per_page: 100
         }
-        with_cache('user_repos') do
-          request('/user/repos', params)
+        with_cache(:user_repos) do
+
+          page_count = get_total_page_for_request('/user/repos', params)
+          if page_count != nil
+            all_user_repos = Array.new
+            write = true
+            (1..page_count).step(1) do |n|
+              params[:page] = n
+              part_res = request('/user/repos', params)
+              if not part_res.is_a?(Net::HTTPSuccess)
+                write = false
+                break
+              else
+                all_user_repos = all_user_repos+ deserialize_body(part_res.body)
+              end
+            end
+            {write: write, cache_content: all_user_repos}
+          else
+            res = request('/user/repos', params)
+            if not res.is_a?(Net::HTTPSuccess)
+              { write: false, cache_content: nil }
+            else
+              { write: true, cache_content: deserialize_body(res.body) }
+            end
+          end
+
         end
+
       end
 
       def user_pulls
-        modifiers = org_modifiers('is:pr', 'user:@me', 'state:open', 'involves:@me')
+
+        if @pr_all_involve_me.nil?
+          modifiers = org_modifiers('is:pr', "user:#@me_account", 'state:open', "involves:#@me_account")
+        else
+          modifiers = ['is:pr', 'state:open', "involves:#@me_account"]
+        end
         params = search_params('', modifiers).merge(
-          per_page: 100
+            per_page: 100
         )
-        response = with_cache('user_pulls') do
-          request('/search/issues', params)
+        response = with_cache(:user_pulls) do
+          page_count = get_total_page_for_request('/search/issues', params)
+          if page_count != nil
+            all_user_pulls = Array.new
+            write = true
+            (1..page_count).step(1) do |n|
+              params[:page] = n
+              part_res = request('/search/issues', params)
+              if not part_res.is_a?(Net::HTTPSuccess)
+                write = false
+                break
+              else
+                all_user_pulls = all_user_pulls+ deserialize_body(part_res.body)
+              end
+            end
+            {write: write, cache_content: all_user_pulls}
+          else
+            res = request('/search/issues', params)
+            if not res.is_a?(Net::HTTPSuccess)
+              { write: false, cache_content: nil }
+            else
+              { write: true, cache_content: deserialize_body(res.body) }
+            end
```

Comment:
This block seems identical to the one in the other method (`user_repos`). Do you think we can refactor it? I would suggest avoid using too many `if` branches as well. You can use early returns to achieve that ;)

Reply:
Thanks for reviewing @edgarjs , I updated the PR with some refactoring and unit tests.

I'm not super familiar with the ruby language feature, open to any suggestion change. Thanks again.